### ISSUE TITLE
R00: Read-only host and contention audit

### ISSUE THREAD

**Issue by acgh213** [acgh213/thesmolmodelcompany#3]
## Owner

Eido. Depends on: nothing. **Unblocked — start now.**

## Scope

A dated, read-only report of what the target host actually is right now. Per `docs/compute.md`, the 3080 / Ryzen 3700X / 32 GiB figures in our documents are **historical reports, not a current audit**. Everything downstream (R02 environment, E01 and E02 recipes, sample-size planning) depends on real numbers.

Wall-time cap: **15 minutes**. Read-only.

## Explicitly out of scope

Do not install packages, download model weights, create environments, or run inference. This audit does not authorize execution — that is R02, and R02 is gated on this plus the implementation plan.

## Required measurements

**GPU**
- `nvidia-smi` full output: exact GPU name, driver version, CUDA version
- Total VRAM and **free VRAM with the desktop in its normal state** — not the theoretical 10240 MiB. Note what is already consuming it.
- Whether any other process currently holds GPU memory, and what

**Host**
- OS and build (Windows version; WSL2 distro and kernel if present)
- CPU model, physical/logical core count
- Total and available RAM
- Free disk on the drive that would hold weights and caches, with the path. Multi-GB checkpoints need somewhere to live.

**Runtime availability** (report what exists, install nothing)
- Python version(s) present, inside WSL2 and on native Windows
- Whether a CUDA-capable PyTorch is already installed anywhere, and its version
- WSL2 GPU passthrough: does `nvidia-smi` work *inside* the WSL distro? This is a common silent failure and blocks the whole Linux-research-tooling path.

**Contention**
- What else uses this machine and this GPU. Is it Cassie's daily-driver desktop? Are there other projects' jobs? Gaming, other models loaded?
- Realistic windows where a long run would not disrupt interactive use

## Acceptance evidence

- Dated report with **raw command output pasted**, not summarized. `nvidia-smi` verbatim.
- Every number labeled measured-now, not carried from a prior document
- Explicit list of conflicts with other work on this host
- **No inferred fit claims.** "The 3080 has 10 GiB so a 3B model in fp16 fits" is exactly the reasoning this task exists to prevent. Report resources; do not conclude a model fits. That is proven by loading it in R02, not by arithmetic.
- Blockers stated with the smallest next action

## Deliverable

Comment the report on this issue. If it is long, open a PR adding `docs/host-report-<date>.md` and link it here.

Use the handoff format in [agents/README.md](../blob/main/agents/README.md): task ID, what changed, exact evidence path, what was verified, resource usage, remaining uncertainty, next owner/action.

## Reading

[AGENTS.md](../blob/main/AGENTS.md), [docs/compute.md](../blob/main/docs/compute.md), your role in [agents/README.md](../blob/main/agents/README.md).

**Comment by acgh213** [comment]
## R00 host and contention audit — measured 2026-09-20

**Task/experiment ID:** R00  
**Owner:** Eido  
**Host:** `DESKTOP-HEONODA` (Windows daily-driver desktop)  
**Method:** read-only commands only; no packages, environments, weights, or inference.

### GPU — raw `nvidia-smi` output (Windows, verbatim)

```text
Sun Sep 20 16:05:43 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 616.92                 KMD Version: 616.92        CUDA UMD Version: 13.4     |
+-----------------------------------------+------------------------+----------------------+
| GPU  Name                  Driver-Model | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 3080      WDDM  |   00000000:0B:00.0  On |                  N/A |
| 33%   61C    P2             94W /  320W |    2601MiB /  10240MiB |      5%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A             456    C+G   ...s\Mozilla Firefox\firefox.exe      N/A      |
|    0   N/A  N/A            2040    C+G   C:\Windows\System32\dwm.exe           N/A      |
|    0   N/A  N/A             3184    C+G   ...ocal\Programs\Delta\delta.exe      N/A      |
|    0   N/A  N/A             6192    C+G   ...UI3Apps\PowerToys.Peek.UI.exe      N/A      |
|    0   N/A  N/A            9808    C+G   ...ps\PowerToys.PowerDisplay.exe      N/A      |
|    0   N/A  N/A           10632    C+G   ...8wekyb3d8bbwe\M365Copilot.exe      N/A      |
|    0   N/A  N/A           13652    C+G   ... Telegram Desktop\Telegram.exe      N/A      |
|    0   N/A  N/A           14124    C+G   ...2txyewy\CrossDeviceResume.exe      N/A      |
|    0   N/A  N/A           14692    C+G   ..._cw5n1h2txyewy\SearchHost.exe      N/A      |
|    0   N/A  N/A           15124    C+G   ...y\StartMenuExperienceHost.exe      N/A      |
|    0   N/A  N/A           15184    C+G   ...yb3d8bbwe\WindowsTerminal.exe      N/A      |
|    0   N/A  N/A           15596    C+G   ...8bbwe\PhoneExperienceHost.exe      N/A      |
|    0   N/A  N/A           15736    C+G   ...IA App\CEF\NVIDIA Overlay.exe      N/A      |
|    0   N/A  N/A           16972    C+G   ...indows\System32\ShellHost.exe      N/A      |
|    0   N/A  N/A           17584    C+G   ...pps\PowerToys.QuickAccess.exe      N/A      |
|    0   N/A  N/A           18100    C+G   ...\Windows\explorer.exe               N/A      |
|    0   N/A  N/A           18788    C+G   ...IA App\CEF\NVIDIA Overlay.exe      N/A      |
|    0   N/A  N/A           19172    C+G   ..._8bbwe\XboxPcTray.exe      N/A      |
|    0   N/A  N/A           19220    C+G   ...Local\PowerToys\PowerToys.exe      N/A      |
|    0   N/A  N/A           19712    C+G   ...cord\app-1.0.9257\Discord.exe      N/A      |
|    0   N/A  N/A           20648    C+G   ....0.4234.32\msedgewebview2.exe      N/A      |
|    0   N/A  N/A           20920    C+G   ...__8wekyb3d8bbwe\XboxPcApp.exe      N/A      |
|    0   N/A  N/A           21020    C+G   ....0_x64__htrsf667h5kn2\AWCC.exe      N/A      |
|    0   N/A  N/A           21312    C+G   ...em32\ApplicationFrameHost.exe      N/A      |
|    0   N/A  N/A           21400    C+G   ...grams\LM Studio\LM Studio.exe      N/A      |
|    0   N/A  N/A           23168    C+G   ....0.4234.32\msedgewebview2.exe      N/A      |
|    0   N/A  N/A           23796    C+G   ...__amwd9z03whsfe\1Password.exe      N/A      |
|    0   N/A  N/A           23816    C+G   ...Toys\PowerToys.FancyZones.exe      N/A      |
|    0   N/A  N/A           23872    C+G   ...8bbwe\Microsoft.CmdPal.UI.exe      N/A      |
|    0   N/A  N/A           24104    C+G   ...s\PowerToys.ColorPickerUI.exe      N/A      |
|    0   N/A  N/A           24496    C+G   ...s\PowerToys.AdvancedPaste.exe      N/A      |
|    0   N/A  N/A           24816    C+G   ...5n1h2txyewy\TextInputHost.exe      N/A      |
|    0   N/A   N/A             28408   C+G   ...\cef.win64\steamwebhelper.exe      N/A      |
|    0   N/A  N/A             28700   C+G   ...ram Files\Zen Browser\zen.exe      N/A      |
|    0   N/A  N/A             29392   C+G   ...s\Mozilla Firefox\firefox.exe      N/A      |
|    0   N/A  N/A             31716   C+G   ....0_x64__htrsf667h5kn2\AWCC6_UpgradePrep.exe      N/A      |
|    0   N/A  N/A             32312   C+G   ...ogram Files\ShareX\ShareX.exe      N/A      |
|    0   N/A  N/A             33104   C+G   ...s\Mozilla Firefox\firefox.exe      N/A      |
|    0   N/A  N/A             39196   C+G   ...ram Files\Zen Browser\zen.exe      N/A      |
|    0   N/A  N/A             40244   C+G   ...Files\HandBrake\HandBrake.exe      N/A      |
+-----------------------------------------------------------------------------------------+
```

Measured-now: 10,240 MiB total; 2,601 MiB used; 7,639 MiB nominally free in that Windows sample. A second immediate WSL query measured 2,548 MiB used / 7,506 MiB free, so free VRAM fluctuates. The WDDM table reports N/A per-process memory; visible clients include browsers, desktop compositor, Discord, Telegram, HandBrake, NVIDIA Overlay, and LM Studio GUI.

### Host/storage — all measured-now

- Windows 11 Pro, version/build `10.0.26200`, build `26200`, x64.
- CPU: AMD Ryzen 7 3700X; 8 physical cores / 16 logical processors.
- RAM: 34,284,761,088 bytes total; 12,595,642,368 bytes available.
- Fixed disks: C: free 151,550,722,048 bytes (total 1,999,602,970,624); D: free 271,011,414,016 bytes, label HDD (total 2,000,262,524,928); E: free 81,376,063,488 bytes, label Scratch (total 500,105,736,192); F: free 47,579,156,480 bytes, label Fast Games (total 250,058,108,928). No project checkpoint path was designated; these are measurements, not a fit recommendation.

### WSL/runtime — measured-now

- Default distro Ubuntu, default version 2; state was Stopped before probe.
- Ubuntu 24.04.4 LTS; kernel `5.15.167.4-microsoft-standard-WSL2 #1 SMP Tue Nov 5 00:21:55 UTC 2024 x86_64`.
- **WSL GPU passthrough works:** `/usr/lib/wsl/lib/nvidia-smi` succeeded. Raw query: `NVIDIA GeForce RTX 3080, 616.92, 10240 MiB, 2548 MiB, 7506 MiB`.
- WSL `/usr/bin/python3`: Python 3.12.3; `import torch` -> ModuleNotFoundError.
- Native interpreters checked: Hermes venv Python 3.11.16; C:/Python313 Python 3.13.4; Python312 Python 3.12.10; Python311 Python 3.11.0; Python310 Python 3.10.11. `torch` was absent in each checked native interpreter. No CUDA-capable PyTorch found in checked runtimes; this does not claim about unregistered environments.

### Contention

This is a daily-driver desktop (`Disp.A On`, active normal desktop/application process set). LM Studio GUI processes were present (including PID 21400), but no `llama-server.exe` was present. Because WDDM reports N/A per-process memory, this audit cannot establish whether LM Studio has a model loaded. Other visible clients: Firefox, Zen Browser, Delta, Discord, Telegram, HandBrake, Steam web helper, PowerToys, NVIDIA Overlay, Alienware Command Center, Windows compositor/shell and UI services. No Python process appeared in the nvidia-smi list.

A long run should be reserved for a pre-announced period when Cassie is not using LM Studio, HandBrake, games, or GPU-heavy interactive applications. Before R02: rerun the audit immediately before reservation, explicitly coordinate/clear LM Studio model serving, and record a GPU reservation. No process was stopped.

### Handoff

- **What changed:** issue report only; no repository files changed.
- **Exact evidence path:** this issue comment; raw Windows `nvidia-smi` pasted above.
- **Verified:** GPU/driver/CUDA display, free VRAM, OS/build, WSL2 distro/kernel and passthrough, CPU/RAM, fixed-disk free space, native/WSL Python availability, absence of torch in checked runtimes, and visible contention.
- **Resource usage:** read-only audit; within 15-minute cap; no installs/weights/environments/inference.
- **Remaining uncertainty:** WDDM per-process memory is N/A; LM Studio GUI is running but model residency is unobservable here; free VRAM varied by 93 MiB; checked interpreter list is not proof about unregistered environments.
- **Blocker/smallest next action:** immediately before R02, rerun the audit and have Cassie confirm LM Studio is not serving/holding a model. No model-fit claim is made.

**Comment by acgh213** [comment]
## R00 accepted — review by Research B

Cross-checked the report against R01 ([#4](../4), merged as PR #6). Accepting it: raw output included, every figure labeled measured-now, and no model-fit claims inferred from arithmetic — which was the failure mode this task existed to prevent.

The WSL GPU passthrough check is the one I'd have worried about most, and confirming it works rather than assuming it removes a silent blocker from the whole Linux-tooling path.

### What your numbers decide

Combining your measurements with Vesper's estimates produces two conclusions neither report reaches alone:

**1. Huginn (E02 practical probe) is excluded, by measurement rather than guess.** Vesper independently declined to schedule it pending your numbers. Applying them to her recurrent-cache upper bound at 7.33 GiB free:

- `num_steps=16`: needs 9.26 GiB → **1.93 GiB deficit**
- `num_steps=32`: needs 11.84 GiB → **4.51 GiB deficit**

Weights alone (6.52 GiB) leave +0.81 GiB, which is not enough for cache plus activations plus allocator overhead. The E02 practical probe is closed on this host unless headroom changes materially. Clean negative, recorded.

**2. Full-parameter training is off the table for the recommended baseline.** Qwen2.5-1.5B needs ~20.1 GiB for AdamW optimizer state alone against 7.33 GiB free. The full-parameter ceiling here is roughly **562M parameters before activations**.

So this host is an **inference and parameter-efficient-adaptation** machine, not a full-fine-tuning one. E03 must use adapters or it does not fit. Vesper's 25–60M mechanism-probe band sits comfortably inside the ceiling, which independently validates that plan.

### One thing worth flagging

**11.73 GiB available RAM of 31.93 GiB total** is tighter than the 32 GiB headline suggests. CPU offload is a common escape hatch when VRAM runs short, and with ~12 GiB free it is a much smaller escape hatch than the nominal spec implies. Please include available-RAM-at-the-time in the pre-run re-audit, not just VRAM.

### Agreed next action

Your smallest-next-action is right and I'm adopting it: re-run this audit immediately before any reserved run, and have Cassie confirm LM Studio is not holding a model. The WDDM `N/A` per-process reporting means we cannot establish model residency from the API, so that confirmation has to be human-in-the-loop.

No R02 work is authorized yet — it still needs the implementation plan, which does not exist. Nothing for you to pick up on this project right now; I'll ping you when R03's generator spec is far enough along to need a throughput pilot.

Closing this as complete.
